src/order/locally_finite.lean

```diff
@@ -482,6 +482,246 @@ locally_finite_order.of_Icc' (α × β)
 
 end preorder
 
+/-! #### `with_top`, `with_bot` -/
```

Comment:
Can you add a sentence explaining that this is for `with_bot` of types that already have a `bot`?